tests/unit/components/Scorecard.spec.ts

```diff
@@ -0,0 +1,117 @@
+    const wrapper = mount(Scorecard);
+    await wrapper.find('[data-test="share"]').trigger("click");
+
+    expect(mockWindowOpen).toHaveBeenCalled();
+  });
+
+  it("share text on whatsapp when no questions answered", async () => {
+    const wrapper = mount(Scorecard);
+    await wrapper.find('[data-test="share"]').trigger("click");
+
+    expect(mockWindowOpen).toHaveBeenCalledWith(
+      "https://api.whatsapp.com/send/?phone&text=%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%0A%0A%F0%9F%8F%86%20*Hooray!%20I%20completed%20a%20Quiz!*%20%F0%9F%8F%86%0A%0A%F0%9F%8C%9F%20*undefined*%20%F0%9F%8C%9F%0A%0AI%20answered%20undefined%20questions%20with%200%25%20accuracy%20on%20Avanti%20Fellows%20quiz%20today!%20%F0%9F%98%87%0A%0A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A"
+    );
+  });
+
+  it("share text on whatsapp when one question answered", async () => {
+    const progressPercentage = 50;
+    const wrapper = mount(Scorecard);
+    await wrapper.setProps({
+      numQuestionsAnswered: 4,
+      progressPercentage: progressPercentage,
+    });
+    await flushPromises();
+    await jest.advanceTimersByTime(1000);
+
+    await wrapper.find('[data-test="share"]').trigger("click");
+
+    expect(mockWindowOpen).toHaveBeenCalledWith(
+      `https://api.whatsapp.com/send/?phone&text=%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%0A%0A%F0%9F%8F%86%20*Hooray!%20I%20completed%20a%20Quiz!*%20%F0%9F%8F%86%0A%0A%F0%9F%8C%9F%20*undefined*%20%F0%9F%8C%9F%0A%0AI%20answered%204%20questions%20with%200%25%20accuracy%20on%20Avanti%20Fellows%20quiz%20today!%20%F0%9F%98%87%0A%0A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A`
+    );
+  });
+
+  it("share text on whatsapp when multiple questions answered", async () => {
+    const progressPercentage = 50;
+    const wrapper = mount(Scorecard);
+    await wrapper.setProps({
+      numQuestionsAnswered: 4,
+      progressPercentage: progressPercentage,
+    });
+    await flushPromises();
+    await jest.advanceTimersByTime(1000);
+
+    await wrapper.find('[data-test="share"]').trigger("click");
+
+    expect(mockWindowOpen).toHaveBeenCalledWith(
+      `https://api.whatsapp.com/send/?phone&text=%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%0A%0A%F0%9F%8F%86%20*Hooray!%20I%20completed%20a%20Quiz!*%20%F0%9F%8F%86%0A%0A%F0%9F%8C%9F%20*undefined*%20%F0%9F%8C%9F%0A%0AI%20answered%204%20questions%20with%200%25%20accuracy%20on%20Avanti%20Fellows%20quiz%20today!%20%F0%9F%98%87%0A%0A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A%F0%9F%8E%89%F0%9F%8E%8A`
+    );
+  });
+});
```

Comment:
you can test this one as well by mocking the entire `dom-to-image` module - you can read up how to mock modules in jest.
<img width="541" alt="Screenshot 2022-04-08 at 3 00 49 PM" src="https://user-images.githubusercontent.com/13185699/162408385-6158ec65-54f8-4a20-a372-8a52dca3c42f.png">